src/Build/BuildCheck/Infrastructure/CustomConfigurationData.cs

```diff
@@ -55,10 +65,40 @@ public override bool Equals(object? obj)
             return false;
         }
 
-        return Equals((CustomConfigurationData)obj);
-    }
+        var customConfigObj = (CustomConfigurationData) obj;
+
+        if(customConfigObj.RuleId != RuleId)
+        {
+            return false;
+        }
 
-    protected bool Equals(CustomConfigurationData other) => Equals(ConfigurationData, other.ConfigurationData);
+        // validate keys and values
+        if (customConfigObj.ConfigurationData != null && ConfigurationData != null)
+        {
+            if (!customConfigObj.ConfigurationData.Keys.SequenceEqual(ConfigurationData.Keys))
```

Comment:
This doesn't look correct. I don't think the order of enumeration of `Keys` is defined.

Reply:
Thank you for catching it. Added test to verify it: TestCustomConfigurationData_Equals_ShouldBeTrue_CustomConfigDataKeysOrderDiffers()
and updated the implementation.